tests/operations/test_operations.py

```diff
@@ -51,3 +53,22 @@ def test_bitwise_lshift(a, b):
 )
 def test_bitwise_rshift(a, b):
     return a >> b
+
+
+@mt(
+    run(np.arange(10), result=np.prod(np.arange(10))),
+    run(np.asarray([-2, -1, 2, 11], dtype='float32'), result=np.float32(44)),
+)
+def test_prod(arr):
+    return np.prod(arr)
+
+
+@mt(
+    run((2, 3), 0, None, result=np.zeros((2, 3), 'float32')),
```

Comment:
You can import `Shp` from `.common` and use `Shp(2, 3)`. That will create a u64 tuple.

Reply:
Ok ! So, i am using `Shp`, but now I have issues because op_full now receives an abstract type of abstract tuple, instead of just an abstract tuple, so I am looking for a way to fix it.

Reply:
Oh yeah my bad, `Shp` is for for `infer(...)`. You can use `(np.uint64(2), np.uint64(3))`, for what it's worth.